src/main/webapp/app/exercises/programming/participate/programming-submission.service.ts

```diff
@@ -217,6 +303,78 @@ export class ProgrammingSubmissionService implements IProgrammingSubmissionServi
         }
     }
 
+    private setupWebsocketSubscriptionForSubmissionProcessing(participationId: number, exerciseId: number, personal: boolean): void {
+        if (!this.submissionProcessingTopicsSubscribed.get(participationId)) {
+            let newSubmissionTopic: string;
+            if (personal) {
+                newSubmissionTopic = '/user/topic/submissionProcessing';
+            } else {
+                newSubmissionTopic = this.SUBMISSION_PROCESSING_TEMPLATE_TOPIC.replace('%exerciseId%', exerciseId.toString());
+            }
+
+            // Only subscribe if not subscription to same topic exists (e.g. from different participation)
+            if (!Array.from(this.submissionProcessingTopicsSubscribed.values()).includes(newSubmissionTopic)) {
```

Comment:
I think this comment should rather be a variable name :) 

```suggestion
            // e.g. from different participation
            const subscriptionOnSameTopicExists = Array.from(this.submissionProcessingTopicsSubscribed.values()).includes(newSubmissionTopic)
            if (!subscriptionOnSameTopicExists) {
```